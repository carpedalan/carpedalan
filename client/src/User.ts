export enum User {
  read = 'read',
  write = 'write',
  none = 'none',
}
