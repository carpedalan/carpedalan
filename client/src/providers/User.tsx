import * as React from 'react';

export type User = 'read' | 'write' | undefined;
const user: User = window.__SESSION__ ? window.__SESSION__.user : undefined;
const { createContext, Children, useState } = React;

type SetUser = (u: User) => void;
export interface IUserContext {
  user: User;
  setUser: SetUser;
}
export const UserContext = createContext<IUserContext>({
  user,
  setUser: () => {},
});

export const UserConsumer = UserContext.Consumer;

export const UserProvider: React.FC<React.ReactNode> = ({ children }) => {
  const [userState, setUser] = useState<User>(user);
  const value: IUserContext = {
    setUser,
    user: userState,
  };
  return (
    <UserContext.Provider value={value}>
      {Children.only(children)}
    </UserContext.Provider>
  );
};
