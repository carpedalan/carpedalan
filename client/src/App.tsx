import axios from 'axios';
import debug from 'debug';
import { default as useUser } from 'hooks/useUser';
import { DataProvider } from 'providers/Data';
import * as React from 'react';
import { BrowserRouter } from 'react-router-dom';
import Routes from 'Routes';
import { ThemeProvider } from 'styled-components';
import { GlobalStyleComponent, themes } from 'styles/utils';
import { User } from 'User';

const log = debug('App');

const refreshCookie = async () => {
  log('Refreshing cookiaxdae');
  const response = await axios.post('/v1/refresh');
};

const { useEffect } = React;
const App: React.FC<{ user: User }> = () => {
  const { user: userState } = useUser();

  /**
   * Cloudfront cookie needs to be refreshed every 30s for
   * the images to not 403. This effect well set the cookie at
   * that interval if the user has been authenticated.
   */
  useEffect(() => {
    if (userState) {
      const interval = setInterval(() => {
        refreshCookie();
      }, 1000 * 31);
      return () => clearInterval(interval);
    }
    return () => {};
  }, [userState]);

  return (
    <DataProvider>
      <BrowserRouter>
        <ThemeProvider theme={themes.lite}>
          <>
            <Routes />
            <GlobalStyleComponent />
          </>
        </ThemeProvider>
      </BrowserRouter>
    </DataProvider>
  );
};

export default App;
