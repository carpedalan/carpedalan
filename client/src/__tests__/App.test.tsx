import { shallow } from 'enzyme';
import * as React from 'react';
import App from '../App';

import { User } from '../User';

describe('<App />', () => {
  it('should match snapshot', () => {
    const app = shallow(<App user={User.read} />);
    expect(app).toMatchSnapshot();
  });
  it('should increment', () => {
    const app = shallow(<App user={User.read} />);
    const something = app.find('[data-test="something"]');
    something.simulate('click');
    expect(app.find('[data-test="something"]').text()).toContain(1);
  });
});
