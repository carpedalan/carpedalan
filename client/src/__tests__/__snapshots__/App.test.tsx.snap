// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match snapshot 1`] = `
<BrowserRouter>
  <Link
    to="login"
  >
    Login
  </Link>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/login"
    />
  </Switch>
  <div
    data-test="something"
    onClick={[Function]}
  >
    ffuck yes
    0
  </div>
</BrowserRouter>
`;
